tensorboard/defs/defs.bzl

```diff
@@ -98,50 +100,94 @@ def tf_ts_library(strict_checks = True, **kwargs):
 
     if strict_checks == False:
         tsconfig = "//:tsconfig-lax"
-    elif "test_only" in kwargs and kwargs.get("test_only"):
-        tsconfig = "//:tsconfig-test"
     kwargs.setdefault("deps", []).extend(["@npm//tslib", "//tensorboard/defs:strict_types"])
 
-    ts_library(tsconfig = tsconfig, supports_workers = True, **kwargs)
+    new_srcs = []
+    # Find test.ts files and rename to test.spec.ts to be compatible with
+    # spec_bundle() tooling.
+    for s in srcs:
+      if s.endswith("_test.ts") or s.endswith("-test.ts"):
+        # Make a copy of the file with name .spec.ts and switch to that as
+        # the src file.
+        new_src = s[0:s.rindex('.ts')] + ".spec.ts"
+        native.genrule(
```

Comment:
Am I correct that this rule copies the contents of file s into the new_src file? Would a cp command make more sense here? I guess if this works lets leave it. It just seems strange to cat the contents into the file.

Reply:
You're right, the `cat` is weird. Actually I discovered there is a `copy_file()` rule:
https://github.com/bazelbuild/bazel-skylib/blob/main/docs/copy_file_doc.md

It will supposedly just use symlinks on UNIX so maybe that's even better.

Reply:
It is also easier to read. I like this much better.